src/msgraph_core/requests/batch_request_builder.py

```diff
@@ -60,11 +59,6 @@ async def post(
 
         if isinstance(batch_request_content, BatchRequestContent):
             request_info = await self.to_post_request_information(batch_request_content)
-            bytes_content = request_info.content
-            json_content = bytes_content.decode("utf-8")
-            updated_str = '{"requests":' + json_content + '}'
-            updated_bytes = updated_str.encode("utf-8")
-            request_info.content = updated_bytes
```

Comment:
I believe the fix looks correct(the builder should not be trying to buildup/modify already serialized JSON content). 
But I'm not sure how this got checked into the `main` branch as this was something that was brought up in this PR IIRC.
https://github.com/microsoftgraph/msgraph-sdk-python-core/pull/702#discussion_r1814828420

Reply:
It's intentional. The logic was redundant because when building the request information object in line 61,
`setContentFromParsable()` calls `BatchRequestContent`'s `serialize()` which writes its contents to the stream.